Sources/FusionMedia_Android/AudioPlayer+Android.swift

```diff
@@ -5,31 +5,71 @@ import FusionMedia_Common
 
 
 public class AudioPlayer {
-  let url: URL
-  var player: MediaPlayer? = nil 
+  	let url: URL
+  	var player: MediaPlayer? = nil
+  	var volume: Float = 1.0
+   
+  	private var listener: MediaPlayerListener? = nil
   
-  private var listener: MediaPlayerListener? = nil
-  
-  public required init(url: URL) {
-    self.url = url
-  }     
+  	public required init(url: URL) {
+    	self.url = url
+    	self.player = MediaPlayer()
+    	self.listener = MediaPlayerListener()
+    
+    	self.player?.setVolume(leftVolume: volume, rightVolume: volume)            
+    	self.player?.setOnPreparedListener(listener: listener)
+    	self.player?.setDataSource(path: url.absoluteString)
+    	self.player?.prepareAsync()
+  	}     
 }
 
-extension AudioPlayer: AudioPlayerProtocol {
-  public func play() {
-    guard let isPlaying = self.player?.isPlaying(), !isPlaying else { return }
+extension AudioPlayer: AudioPlayerProtocol {    
+  	public func play() {
+    	guard let player = self.player, !player.isPlaying() else { return }
+
+		player.start()
+  	}
+  
+  	public func stop() {
+        guard let player = player else { return }
+   	 	if player.isPlaying() {
+      		player.pause()
+    	}
+  	}
+  
+  	public func isPlaying() -> Bool {
+        guard let player = player else { return false }  	
+  		return player.isPlaying()
+  	}
+  
+    public func getDuration() -> Double {
+        guard let player = player else { return 0 }
+        return Double(player.getDuration())
+    }
     
-    self.player = MediaPlayer()
-    self.listener = MediaPlayerListener()        
-            
-    self.player?.setOnPreparedListener(listener: listener)
-    self.player?.setDataSource(path: url.absoluteString)
-    self.player?.prepareAsync()
-  }
+    public func setProgress(_ progress: Float) {
```

Comment:
Please, create the API in more functional style, e.g. function for an action/var for a status/state/etc. Try to avoid set/get combination. Best ist just to make the API as similar as possible to the Apple's API as it's originally designed to be used with Swift. In this case, the Apple's API is also quite similar to the Android:

var duration { get }
var currentTime { get }
var currentPosition { get }
var volume { get set }

func seek(to:)

...